cocos/editor-support/cocostudio/CCColliderDetector.cpp

```diff
@@ -196,7 +196,7 @@ void ColliderDetector::addContourData(ContourData *contourData)
     unsigned long num = contourData->vertexList.size();
     for (unsigned long i = 0; i < num; i++)
     {
-        calculatedVertexList.push_back(Vec2());
+        calculatedVertexList.emplace_back();
```

Comment:
May be just use reserve here?

Reply:
Yes. `reserve` can be used here. However, upon reading the context around this line, I found it hard to understand.

`calculatedVertexList` is a reference to a member,

```cpp
std::vector<Vec2> &calculatedVertexList = colliderBody->_calculatedVertexList;
```

It's initialized in

```cpp
#if ENABLE_PHYSICS_SAVE_CALCULATED_VERTEX
    _calculatedVertexList = Array::create();
    CC_SAFE_RETAIN(_calculatedVertexList);
#endif
```

Does the code in the macro ever run? I am not familiar with the `ColliderBody` class. The following line wouldn't compile.

```cpp
std::vector<cocos2d::Vec2> vertex_list = Array::create();
```

Reply:
I am not familiar with it too. But the codes is `calculatedVertexList.emplace_back();` is not easy to understand. It seems it emplace back nothing.

Reply:
`emplace_back()` means default constructor being used.

Reply:
Yep, i understand.